### PR TITLE
ingest-node.asciidoc - Clarify json processor

### PR DESCRIPTION
**Pull request by GlenRSmith**
Clarify documentation of json processor with a config, a sample doc, and the processed result

### CODE REVIEW

**Comment by clintongormley** [comment]
@GlenRSmith LGTM, would be even nicer if these snippets used runnable CONSOLE snippets and included the docs tests (so we would actually confirm that the snippets run and that they return the result we're expecting)  See https://github.com/elastic/elasticsearch/blob/master/docs/README.asciidoc for details

**Comment by GlenRSmith** [comment]
@clintongormley Thanks for the explanatory link. I knew that feature was in play now, but nothing about implementation. This will be super helpful. But this PR failed some integration test?

**Comment by jasontedor** [comment]
> But this PR failed some integration test?

It was an unrelated failure, fixed by #21894.

**Comment by jasontedor** [comment]
test again please

**Comment by GlenRSmith** [comment]
test again please

**Comment by jasontedor** [comment]
test again please

**Comment by jasontedor** [comment]
retest this please

**Comment by GlenRSmith** [comment]
retest this please

**Comment by nik9000** [comment]
@GlenRSmith, would you like to fix the merge conflict?

File: docs/reference/ingest/ingest-node.asciidoc [hunk 1]
```diff
@@ -1510,32 +1510,83 @@ Converts a JSON string into a structured JSON object.
...
+--------------------------------------------------
+
+This illustrates that, unless it is explicitly named in the processor configuration, the `target_field` 
+is the same field provided in the required `field` configuration.
+
 [[kv-processor]]
 === KV Processor
 This processor helps automatically parse messages (or specific event fields) which are of the foo=bar variety.
 
 For example, if you have a log message which contains `ip=1.2.3.4 error=REFUSED`, you can parse those automatically by configuring:
```
Comment (GlenRSmith) [inline docs/reference/ingest/ingest-node.asciidoc]: Need to restore all the relevant markup around the examples in the kv processor.

**Comment by GlenRSmith** [comment]
retest this please

**Comment by GlenRSmith** [comment]
Should pass now, but still needs snippets appropriately tagged for CONSOLE and TEST as commented by @clintongormley previously.

**Comment by nik9000** [comment]
I'd love to convert the snippets, yeah, but I don't think we need to block the merge on it. If the rest of the file were all CONSOLE and you were adding non-CONSOLE snippets, sure. If you think you'll have a chance to convert them then let me know and I'll wait. Otherwise I think we should merge to get the better docs and convert when we get around to it.

I've been slowly converting the unconverted snippets and I'd get to these eventually.

**Comment by GlenRSmith** [comment]
@nik9000 Then kindly merge away. I don't know when I would get to converting, and if I do get to it, well, it will be sitting there waiting.

**Comment by nik9000** [comment]
OK @GlenRSmith! I've merged and cherry-picked to 5.x, 5.4, and 5.3.

**Pull request merged**